### ISSUE TITLE
CFT re-write

### ISSUE THREAD

**Issue by vin0110** [vin0110/haas#36]
Fix CFT per Chris King's suggestions.

Remove autoscaling group.

@oxhead What else?

**Comment by oxhead** [comment]
These are possible items
* Remove autoscaling group
* Auto recovery
* Replace the launch script path with [auto_hpcc.sh](https://github.com/vin0110/haas/blob/master/scripts/auto_hpcc.sh)

**Comment by oxhead** [comment]
Using autoscaling group is a better solution because 

> an instance is a resource and therefore, creating multiple instances in CFT requires manual generating multiple entries

The [document](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-instance.html) does not indicate any way to specify the count.  Same as in the [AWS forum](https://forums.aws.amazon.com/message.jspa?messageID=338188).
We may write a program that generates the required CFT.  But I don't think it's a good way.  To the best of my knowledge, there is no better way than AutoScaling for creating a set of instances

**Comment by vin0110** [comment]
Please check the above with Chris. If he was wrong he would like to know that.

**Comment by vin0110** [comment]
BTW, I confirmed there is [no charge](https://aws.amazon.com/autoscaling/pricing/) for auto scaling.

**Comment by oxhead** [comment]
The **auto recovery** is triggered by CloudWatch.  In order to make this work, we need to associate an alarm with an instance.  In such a case, we cannot add **auto recovery** to instances in an autoscaling group because they are not initialized yet.

We can configure **auto recovery** for each instance after the stack is created.  We can do it on the web portal or via CLI.

However, it seems the **auto recovery** has its limitation: it is triggered only by **StatusCheckFailed_System**.  This type of failure cannot be simulated ([see here](https://forums.aws.amazon.com/thread.jspa?threadID=173823)).

In short, I don't know yet how to verify whether **auto recovery** works on HPCC.

**Comment by oxhead** [comment]
A Thor job cannot complete after a Thor slave failed immediately.  Therefore, **auto recovery** does not help resume a failed job.

When a slave failed (killed the Thor slave process), the Thor master starts a new slave process after few seconds.  In such a case, we might not need to restart the Thor cluster manually.

Regarding Roxie, there is no master Roxie.  Therefore, we need to restart the failed Roxie process manually.

**Comment by vin0110** [comment]
Can we close this?

**Comment by oxhead** [comment]
Yes. I am closing.
